core/src/main/java/hudson/model/Actionable.java

```diff
@@ -91,18 +95,51 @@
      */
     @Exported(name="actions")
     public final List<? extends Action> getAllActions() {
-        List<Action> _actions = new ArrayList<Action>(getActions());
-        for (TransientActionFactory<?> taf : ExtensionList.lookup(TransientActionFactory.class)) {
-            if (taf.type().isInstance(this)) {
-                try {
-                    _actions.addAll(createFor(taf));
-                } catch (Exception e) {
-                    LOGGER.log(Level.SEVERE, "Could not load actions from " + taf + " for " + this, e);
+        List<Action> _actions = getActions();
+        boolean adding = false;
+        synchronized (Actionable.class) {
```

Comment:
I doubt that it matters, since after the cache is initialized during startup the code will just be doing a null check and the JVM is good at optimizing away contention on monitors in simple cases, but if it makes you happier I can switch to a resource holder pattern.